### PR TITLE
Add mapping of OSM controller version to annotation supported

### PR DESCRIPTION
**Pull request by johnsonshi**
Add mapping of OSM controller version to annotations supported

The PR is stacked on #13

Signed-off-by: Johnson Shi <Johnson.Shi@microsoft.com>

### CODE REVIEW

File: pkg/osm/annotations.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package osm
+
+// SupportedAnnotations maintains a mapping of OSM version to supported annotations.
+var SupportedIngress = map[ControllerVersion][]Annotation{
+	"v0.5": {},
```
Comment (johnsonshi) [inline pkg/osm/annotations.go:5]: still compiling these by prowling the release page

File: pkg/osm/annotations.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+// SupportedAnnotations maintains a mapping of OSM version to supported annotations.
+var SupportedAnnotations = map[ControllerVersion][]Annotation{
+	"v0.5": {},
+	"v0.6": {},
+	"v0.7": {},
+	"v0.8": {},
+	"v0.9": {
+		// Source: https://github.com/openservicemesh/osm/releases/tag/v0.9.0
+		"openservicemesh.io/outbound-port-exclusion-list"},
+	"v0.9.1": {
```
Comment (draychev) [inline pkg/osm/annotations.go:12]: I'd recommend we keep this at `v0.9` - major + minor and skip the patch.  By convention we'd expect no significant changes (no new annotations etc. added) between patch versions.

Comment (johnsonshi) [reply pkg/osm/annotations.go:12]: @draychev quick question - if an annotation is introduced in v0.9.1 for example, should the support matrix in this table be `v0.9` or `v0.10`?

File: pkg/osm/annotations.go (hunk 1 above)
Comment (draychev) [reply pkg/osm/annotations.go:5]: @johnsonshi you could use a script like this:
```bash
for x in $(seq 5 9); do git checkout "upstream/release-v0.${x}"; echo "\n----\nv0.$x"; grep -Iir 'Annotation' $(find . -type f -name '*.go' | grep -v _test.go) | grep " = " | grep 'openservicemesh.io/'; done; git checkout main
```

**Comment by draychev** [comment]
Something like this could help automate this extraction of versions per OSM Controller version:
```bash
for x in $(seq 5 9); do git checkout "upstream/release-v0.${x}"; echo "\n----\nv0.$x"; grep -Iir 'Annotation' $(find . -type f -name '*.go' | grep -v _test.go) | grep " = " | grep 'openservicemesh.io/'; done; git checkout main
```

File: pkg/osm/annotations.go (hunk 2 above)
Comment (draychev) [reply pkg/osm/annotations.go:12]: If an annotation is introduced in v0.9.1 --- then our [release-v0.9](https://github.com/openservicemesh/osm/tree/release-v0.9) branch will have it.

**Comment by johnsonshi** [comment]
Ran Del's helpful script to extract osm annotations per version - I think this can be merged now.

**Comment by johnsonshi** [comment]
Need to add checks for supported annotations like https://github.com/openservicemesh/osm-health/pull/24

File: pkg/osm/types.go [hunk 3]
```diff
@@ -14,3 +14,6 @@ type TrafficSplitVersion string
 
 // HTTPRouteVersion is a string type alias for the SMI HTTPRoute version supported.
 type HTTPRouteVersion string
+
+// Annotation is a string type alias for the an osm annotation.
```
Comment (shalier) [inline pkg/osm/types.go:18]: // Annotation is a string type alias for the osm annotation.

Comment (johnsonshi) [reply pkg/osm/types.go:18]: Good catch!

**Review by draychev — APPROVED** [review]

File: pkg/osm/types.go (hunk 3 above)
Comment (johnsonshi) [reply pkg/osm/types.go]: Resolved.

**Pull request merged**